src/client/lib/hooks/useRecaptcha.tsx

```diff
@@ -118,14 +118,7 @@ const useRecaptcha: UseRecaptcha = (
 
   // We can't initialise recaptcha if no site key or render element is provided.
   if (siteKey === '' || renderElement === '') {
-    return {
-      token,
-      error: true,
-      expired,
-      widgetId,
-      executeCaptcha: () => null,
-      requestCount,
-    };
+    throw new Error('No site key or render element passed');
```

Comment:
I've changed this to throw an error now, because by returning early we were running the `useRecaptchaScript` conditionally on re-render, which goes against the rules of using hooks

Reply:
also updated the test to check for the exception occuring